### PR TITLE
[TAS-2009] ✨ Add address copy feature

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2024-10-17 下午6 09 21

### CODE REVIEW

File: components/AuthStateView.vue [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
+          <UButton
+            class="text-xs font-mon"
+            :label="shortenWalletAddress(wallet)"
+            :to="portfolioURL"
+            variant="soft"
+            block
+            target="_blank"
+          />
+        </UTooltip>
+        <UTooltip text="copy address">
```
Comment (nwingt) [inline components/AuthStateView.vue:15]:
```suggestion
        <UTooltip text="Copy address">
```

Comment (nwingt) [inline components/AuthStateView.vue:15]:
```suggestion
        <UTooltip text="Copy address">
```

File: components/AuthStateView.vue [hunk 2]
```diff
@@ -94,4 +107,13 @@ function onClickDisconnect () {
...
   clearSession()
 }
+
+function onClickCopy () {
+  copyToClipboard(wallet.value)
+  toast.add({
+    icon: 'i-heroicons-clipboard',
+    title: 'Copied to clipboard',
+    timeout: 3000
+  })
```
Comment (nwingt) [inline components/AuthStateView.vue:117]: i think we can move this into `copyToClipboard()`

File: components/AuthStateView.vue [hunk 3]
```diff
@@ -1,16 +1,29 @@
...
+            target="_blank"
+          />
+        </UTooltip>
+        <UTooltip text="copy address">
+          <UButton
+            icon="i-heroicons-document-duplicate"
+            size="sm"
+            color="gray"
+            square
+            variant="link"
```
Comment (nwingt) [inline components/AuthStateView.vue:21]:
```suggestion
            variant="soft"
```

File: components/AuthStateView.vue [hunk 4]
```diff
@@ -1,16 +1,29 @@
...
+            variant="soft"
+            block
+            target="_blank"
+          />
+        </UTooltip>
+        <UTooltip text="copy address">
+          <UButton
+            icon="i-heroicons-document-duplicate"
+            size="sm"
+            color="gray"
```
Comment (nwingt) [inline components/AuthStateView.vue:19]:
```suggestion
```

**Review by nwingt — APPROVED** [review]

File: utils/index.ts [hunk 5]
```diff
@@ -184,3 +184,32 @@ export function convertChannelIdToLikerId (channelId: string) {
...
+  }
+
+  range.selectNode(copyText)
+  selection.removeAllRanges()
+  selection.addRange(range)
+
+  document.execCommand('copy')
+  selection.removeAllRanges()
+
+  document.body.removeChild(copyText)
```
Comment (williamchong) [inline utils/index.ts:208]: Should try `navigator.clipboard` first
`https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText`

**Review by williamchong — CHANGES_REQUESTED** [review]

File: utils/index.ts (hunk 5 above)
Comment (AuroraHuang22) [reply utils/index.ts]: [🧑‍💻 Use navigator.clipboard with fallback](https://github.com/likecoin/nft-book-press/pull/215/commits/b4dca9001dc353b57c9ef5ae71a894d5e06867ab)

**Review by nwingt — COMMENTED** [review]
Maybe just use this https://vueuse.org/core/useClipboard/

File: components/AuthStateView.vue [hunk 6]
```diff
@@ -1,16 +1,28 @@
...
-          :to="portfolioURL"
-          variant="soft"
-          block
-          target="_blank"
-        />
-      </UTooltip>
+      <div class="w-full flex items-center gap-[8px] justify-between">
+        <UTooltip class="flex w-full" :text="wallet">
+          <UButton
+            class="text-xs font-mon"
```
Comment (nwingt) [inline components/AuthStateView.vue:7]:
```suggestion
            class="text-xs font-mono"
```

**Review by nwingt — CHANGES_REQUESTED** [review]

**Pull request merged**